src/snowflake/connector/arrow_iterator.pyx

```diff
@@ -154,20 +163,23 @@ cdef class PyArrowIterator(EmptyPyArrowIterator):
     def __cinit__(self, object cursor, object py_inputstream, object arrow_context, object use_dict_result,
                   object numpy):
         cdef shared_ptr[InputStream] input_stream
-        cdef shared_ptr[CRecordBatchReader] reader
         cdef shared_ptr[CRecordBatch] record_batch
+        cdef CStatus ret
         input_stream.reset(new PyReadableFile(py_inputstream))
-        cdef CStatus ret = CRecordBatchStreamReader.Open(input_stream.get(), &reader)
-        if not ret.ok():
+        cdef CResult[shared_ptr[CRecordBatchReader]] readerRet = CRecordBatchStreamReader.Open(input_stream.get())
```

Comment:
May I ask the reason we go with the new `Open` api here?

Reply:
Arrow changed its API and deprecated the old one. This change now gets rid of warnings and allows us to use versions where this old API was removed.

Reply:
Sounds great!